python-sdk/rebuff/detect_pi_vectorbase.py

```diff
@@ -73,3 +89,55 @@ def init_pinecone(api_key: str, index: str, openai_api_key: str) -> Pinecone:
     vector_store = Pinecone(pc_index, openai_embeddings, text_key="input")
 
     return vector_store
+
+
+if chromadb_installed:
+
+    class ChromaCosineSimilarity(Chroma):
+        """
+        Our code expects a similarity score where similar vectors are close to 1, but Chroma returns a distance score
+        where similar vectors are close to 0.
+        """
+
+        def similarity_search_with_score(
+            self, query: str, k: int, filter=None
+        ) -> List[Tuple[Document, float]]:
+            """
+            Detects Prompt Injection using similarity search with Chroma database.
+
+            Args:
+                query (str): user input to be checked for prompt injection
+                k (int): The threshold for similarity between entries in vector database and the user input.
+
+            Returns:
+                List[Tuple[Document, float]]:  Documents with most similarity with the query and the correspoding similarity scores.
+            """
+
+            results = super().similarity_search_with_score(query, k, filter)
+            return [(document, 1 - score) for document, score in results]
+
+    def init_chroma(
+        url: str, collection_name: str, openai_api_key: str
+    ) -> ChromaCosineSimilarity:
+        """
+        Initializes Chroma vector database.
+
+        Args:
+            url: str, Chroma URL
+            collection_name: str, Chroma collection name
+            openai_api_key (str): Open AI API key
+        Returns:
+            vector_store (ChromaCosineSimilarity)
+
+        """
+
+        openai_embeddings = OpenAIEmbeddings(
+            openai_api_key=openai_api_key, model="text-embedding-ada-002"
+        )
+
+        chroma_collection = ChromaCosineSimilarity(
+            client=chromadb.Client(),
```

Comment:
When `chromadb.Client()` is invoked, is that creating an in-memory database that only persists as long as the process does? If that's the case, it wouldn't really work for a production use case. I think what we'd want to do is use `chromadb.HttpClient` to connect to a remote server. If we go that route, it looks like we might be able to use `chromadb-client` instead which is a more lightweight version of `chromadb` (https://docs.trychroma.com/usage-guide?lang=py#using-the-python-http-only-client).

Reply:
Thank you for the suggestion! I have updated the code to use `chromadb.HttpClient`. I will check if we can also update the dependency to `chromadb-client`